File: web/src/api/team.ts
// M76 — Manager self-service team dashboard + profile timeline.

import { api } from './client'

export interface TeamMember {
  id: string
  employeeNo: string
  firstName: string
  lastName: string
  positionTitle?: string | null
  departmentName?: string | null
  employmentStatus?: string | null
  hireDate?: string | null
  onLeaveToday: boolean
}

export interface UpcomingItem {
  employeeId: string
  date: string
  label: string
}

export interface TeamSummary {
  managerId: string
  headcount: number
  onLeaveToday: number
  onProbation: number
  pendingLeaveRequests: number
  contractsEndingSoon: number
  upcomingProbationReviews: UpcomingItem[]
  upcomingContractEnds: UpcomingItem[]
}

export type TimelineKind =
  | 'HIRE'
  | 'STATUS_CHANGE'
  | 'CONTRACT_SIGNED'
  | 'CONTRACT_ENDED'
  | 'LEAVE_REQUEST'
  | 'BUSINESS_TRIP'
  | 'PERMISSION'
  | 'DISCIPLINARY'
  | 'PROBATION_REVIEW'
  | 'REWARD'
  | 'TERMINATION'
  | 'AUDIT_OTHER'

export interface TimelineEvent {
  at: string
  kind: TimelineKind
  title: string
  detail?: string | null
  actor?: string | null
  referenceId?: string | null
}

export const teamApi = {
  list: () => api.get<TeamMember[]>('/self/team').then((r) => r.data),
  summary: () => api.get<TeamSummary>('/self/team/summary').then((r) => r.data),
}

export const timelineApi = {
  forEmployee: (employeeId: string, limit = 200) =>
    api
      .get<TimelineEvent[]>(`/employees/${employeeId}/timeline`, { params: { limit } })
      .then((r) => r.data),
}

File: mobile/lib/models/team.dart
/// Team/manager models for the mobile app (§11.3).

class TeamMember {
  final String id;
  final String employeeNo;
  final String firstName;
  final String lastName;
  final String? positionTitle;
  final String? departmentName;
  final String employmentStatus;
  final String? hireDate;
  final bool onLeaveToday;

  const TeamMember({
    required this.id,
    required this.employeeNo,
    required this.firstName,
    required this.lastName,
    this.positionTitle,
    this.departmentName,
    required this.employmentStatus,
    this.hireDate,
    required this.onLeaveToday,
  });

  String get fullName => '$firstName $lastName';

  factory TeamMember.fromJson(Map<String, dynamic> j) => TeamMember(
        id: j['id'] as String,
        employeeNo: j['employeeNo'] as String? ?? '',
        firstName: j['firstName'] as String? ?? '',
        lastName: j['lastName'] as String? ?? '',
        positionTitle: j['positionTitle'] as String?,
        departmentName: j['departmentName'] as String?,
        employmentStatus: j['employmentStatus'] as String? ?? '',
        hireDate: j['hireDate'] as String?,
        onLeaveToday: j['onLeaveToday'] as bool? ?? false,
      );
}

class TeamSummary {
  final int headcount;
  final int onLeaveToday;
  final int onProbation;
  final int pendingLeaveRequests;
  final int contractsEndingSoon;

  const TeamSummary({
    required this.headcount,
    required this.onLeaveToday,
    required this.onProbation,
    required this.pendingLeaveRequests,
    required this.contractsEndingSoon,
  });

  factory TeamSummary.fromJson(Map<String, dynamic> j) => TeamSummary(
        headcount: j['headcount'] as int? ?? 0,
        onLeaveToday: (j['onLeaveToday'] as int?) ?? 0,
        onProbation: (j['onProbation'] as int?) ?? 0,
        pendingLeaveRequests: (j['pendingLeaveRequests'] as int?) ?? 0,
        contractsEndingSoon: (j['contractsEndingSoon'] as int?) ?? 0,
      );
}
